src/lib/js/string.rs

```diff
@@ -413,6 +415,81 @@ pub fn last_index_of(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Res
     Ok(to_value(highest_index))
 }
 
+/// Returns an array whose contents is all the results matching the regular expression, if the global (g) flag is present,
+/// in its absence, only the first complete match and its related capturing groups is returned,
+/// otherwise null is returned if no match is found.
+/// <https://tc39.es/ecma262/#sec-string.prototype.match>
+pub fn r#match(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
+    let primitive_val: String = ctx.value_to_rust_string(this);
+    let mut regex_body = String::new();
+    let mut regex_flags = String::new();
+    match args[0].deref() {
+        ValueData::String(ref body) => {
+            // first argument is a string -> use it as regex pattern
+            regex_body = body.into();
+        }
+        ValueData::Object(ref obj) => {
+            let slots = &*obj.borrow().internal_slots;
+            if slots.get("RegExpMatcher").is_some() {
```

Comment:
You could use .get_internal_slot() here, which gives you either the value or `undefined`

Reply:
i think you also want to handle the case where RegExpMatcher isn't there.
(An object was passed but it wasn't a RegExp)